### PR TITLE
🔧 bsx gallery sort & buy now

### LINKED ISSUE #3266: sort in gallery broken
//edit:

sort still needs fine tuning, while the conflicting params don't get passed in at once anymore, it still seems that the sort is not executing correctly (i.e. if you sort for price and then recently interacted, it takes the first argument but not the second one)

~It seems that on beta, with the newest changes, we lost sort. Currently, picking one of the sort options just adds the parameters to the URL i.e.:  https://beta.kodadot.xyz/rmrk/explore?page=1&tab=GALLERY&sort=BLOCK_NUMBER_DESC&sort=UPDATED_AT_ASC&sort=UPDATED_AT_DESC~

image

### PR DESCRIPTION
**Pull request by roiLeo**
⚠️ wait for #3441 before merging this

### PR type

- [x] Bugfix

### What's new?

- [x] PR closes #3266 
- [x] PR related #3293  
- [x] fix "buy now" toggle in `/bsx/explore?page=1&tab=GALLERY`
- [x] fix priceMin & priceMax in `/bsx/explore?page=1&tab=GALLERY`
- [x] fix sort in `/bsx/explore?page=1&tab=GALLERY`

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb)

### Screenshot

Capture d’écran 2022-07-18 à 1 57 37 PM

### CODE REVIEW

File: components/rmrk/Gallery/Search/SearchBar.vue [hunk 1]
```diff
@@ -189,20 +192,25 @@ export default class SearchBar extends mixins(
...
   @Prop(Boolean) public hideSearchInput!: boolean
   @Prop(Boolean) public showDefaultSuggestions!: boolean
 
   protected isVisible = false
+  // private query: SearchQuery = {
+  //   search: '',
+  //   type: '',
+  //   listed: true,
+  //   sortByMultiple: ['BLOCK_NUMBER_DESC'],
+  // }
```
Comment (roiLeo) [inline components/rmrk/Gallery/Search/SearchBar.vue:208]:
```suggestion
```

**Comment by petersopko** [comment]
couple things I've noticed:
- buy now toggle seems to be turned off by default, as opposed to RMRK, i.e. we're showing unlisted nft as default
- when buy now toggle is off (showing unlisted) and I sort NFTs by price high to low, I can still see listed NFTs in the search?
- range slider and setting priceMin + priceMax seems to work, but it need adjustment for bsx
- https://github.com/kodadot/nft-gallery/issues/3461

**Review by petersopko — CHANGES_REQUESTED** [review]
left comment

**Comment by petersopko** [comment]
looks like rmrk/explore/gallery gets wrong query now?
image

```json
{
   "errors":[
      {
         "message":"Variable \"$orderBy\" got invalid value null at \"orderBy[0]\"; Expected non-nullable type \"NftEntitiesOrderBy!\" not to be null.",
         "extensions":{
            "code":"BAD_USER_INPUT",
            "exception":{
               "stacktrace":[
                  "GraphQLError: Variable \"$orderBy\" got invalid value null at \"orderBy[0]\"; Expected non-nullable type \"NftEntitiesOrderBy!\" not to be null.",
                  "    at /usr/local/lib/node_modules/@subql/query/node_modules/graphql/execution/values.js:116:15",
                  "    at coerceInputValueImpl (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/utilities/coerceInputValue.js:57:5)",
                  "    at /usr/local/lib/node_modules/@subql/query/node_modules/graphql/utilities/coerceInputValue.js:70:14",
                  "    at Array.map (<anonymous>)",
                  "    at safeArrayFrom (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/jsutils/safeArrayFrom.js:36:23)",
                  "    at coerceInputValueImpl (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/utilities/coerceInputValue.js:68:50)",
                  "    at coerceInputValue (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/utilities/coerceInputValue.js:37:10)",
                  "    at _loop (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/execution/values.js:109:69)",
                  "    at coerceVariableValues (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/execution/values.js:121:16)",
                  "    at getVariableValues (/usr/local/lib/node_modules/@subql/query/node_modules/graphql/execution/values.js:50:19)"
               ]
            }
         }
      }
   ]
}
```

**Comment by roiLeo** [comment]
@petersopko should be good now

**Comment by petersopko** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 60.37 USD/KSM ~ 1.656 $KSM
🧗 [DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb](https://kusama.subscan.io/account/DVYy1qnocE8t6ZvUfPx3rEjG829khNRXx3YrCGVHHj19Lcb)
🔗 [0xb2d781bb33779dbe7feef064897caa8be09f15b6d65a55e7036244348d682596](https://kusama.subscan.io/extrinsic/0xb2d781bb33779dbe7feef064897caa8be09f15b6d65a55e7036244348d682596)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**